File: app/src/main/java/com/telolahy/guitarmemo/MainActivity.java
package com.telolahy.guitarmemo;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.core.view.MenuItemCompat;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemClickListener {

    private ListView mListView;
    private SongModel mModel;
    private SongAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_layout);
        mModel = new SongModel(this);
        mListView = (ListView) findViewById(R.id.listView1);
        mListView.setOnItemClickListener(this);
        reloadListViewWithAllSongs();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.main_menu, menu);
        MenuItem searchItem = menu.findItem(R.id.action_search);
        final SearchView searchView = (SearchView) MenuItemCompat.getActionView(searchItem);
        searchView.setQueryHint(getResources().getString(R.string.action_search));
        super.onCreateOptionsMenu(menu);
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {
                searchView.clearFocus();
                return false;
            }

            @Override
            public boolean onQueryTextChange(String s) {
                reloadListViewWithSongsContainingKey(s);
                return false;
            }
        });

        // When using the support library, the setOnActionExpandListener() method is
        // static and accepts the MenuItem object as an argument
        MenuItemCompat.setOnActionExpandListener(searchItem, new MenuItemCompat.OnActionExpandListener() {
            @Override
            public boolean onMenuItemActionCollapse(MenuItem item) {
                // Do something when collapsed
                reloadListViewWithAllSongs();
                return true;  // Return true to collapse action view
            }

            @Override
            public boolean onMenuItemActionExpand(MenuItem item) {
                // Do something when expanded
                return true;  // Return true to expand action view
            }
        });

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        switch (item.getItemId()) {
            case R.id.action_overflow:
                break;
            case R.id.refresh:
                break;
            case R.id.chords:
                Intent intent = new Intent(this, ChordsActivity.class);
                startActivity(intent);
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    private void reloadListViewWithAllSongs() {

        mAdapter = new SongAdapter(this, R.layout.item_layout, mModel.allSongs());
        mListView.setAdapter(mAdapter);

    }

    private void reloadListViewWithSongsContainingKey(String key) {

        mAdapter = new SongAdapter(this, R.layout.item_layout, mModel.songsWithKey(key));
        mListView.setAdapter(mAdapter);
    }

    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {

        Song selectedSong = mAdapter.getItem(i);
        Intent intent = new Intent(this, SongActivity.class);
        intent.putExtra(SongActivity.SONG_FILE_EXTRA, selectedSong.file);
        intent.putExtra(SongActivity.SONG_TITLE_EXTRA, selectedSong.title);
        startActivity(intent);
    }
}

File: app/src/main/java/com/telolahy/guitarmemo/MainActivity.kt
package com.telolahy.guitarmemo

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.core.view.MenuItemCompat

class MainActivity : AppCompatActivity(), OnItemClickListener {
    private var mListView: ListView? = null
    private var mModel: SongModel? = null
    private var mAdapter: SongAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_layout)
        mModel = SongModel(this)
        mListView = findViewById<View>(R.id.listView1) as ListView
        mListView!!.onItemClickListener = this
        reloadListViewWithAllSongs()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        val searchItem = menu.findItem(R.id.action_search)
        val searchView = MenuItemCompat.getActionView(searchItem) as SearchView
        searchView.queryHint = resources.getString(R.string.action_search)
        super.onCreateOptionsMenu(menu)
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(s: String): Boolean {
                searchView.clearFocus()
                return false
            }

            override fun onQueryTextChange(s: String): Boolean {
                reloadListViewWithSongsContainingKey(s)
                return false
            }
        })

        // When using the support library, the setOnActionExpandListener() method is
        // static and accepts the MenuItem object as an argument
        MenuItemCompat.setOnActionExpandListener(
            searchItem,
            object : MenuItemCompat.OnActionExpandListener {
                override fun onMenuItemActionCollapse(item: MenuItem): Boolean {
                    // Do something when collapsed
                    reloadListViewWithAllSongs()
                    return true // Return true to collapse action view
                }

                override fun onMenuItemActionExpand(item: MenuItem): Boolean {
                    // Do something when expanded
                    return true // Return true to expand action view
                }
            })

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        when (item.itemId) {
            R.id.action_overflow -> {}
            R.id.refresh -> {}
            R.id.chords -> {
                val intent = Intent(this, ChordsActivity::class.java)
                startActivity(intent)
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun reloadListViewWithAllSongs() {
        mAdapter = SongAdapter(this, R.layout.item_layout, mModel!!.allSongs())
        mListView!!.adapter = mAdapter
    }

    private fun reloadListViewWithSongsContainingKey(key: String) {
        mAdapter = SongAdapter(this, R.layout.item_layout, mModel!!.songsWithKey(key))
        mListView!!.adapter = mAdapter
    }

    override fun onItemClick(adapterView: AdapterView<*>?, view: View, i: Int, l: Long) {
        val selectedSong = mAdapter!!.getItem(i)
        val intent = Intent(this, SongActivity::class.java)
        intent.putExtra(SongActivity.SONG_FILE_EXTRA, selectedSong!!.file)
        intent.putExtra(SongActivity.SONG_TITLE_EXTRA, selectedSong.title)
        startActivity(intent)
    }
}
